kyo-data/shared/src/main/scala/kyo/Schedule.scala

```diff
@@ -243,103 +263,168 @@ object Schedule:
 
         final case class Exponential(initial: Duration, factor: Double) extends Schedule:
-            def next = Maybe((initial, Exponential(initial * factor, factor)))
-            def show = s"Schedule.exponential(${initial.show}, ${formatDouble(factor)})"
+            def next(now: Instant) = Maybe((initial, Exponential(initial * factor, factor)))
+            def show               = s"Schedule.exponential(${initial.show}, ${formatDouble(factor)})"
 
         final case class Fibonacci(a: Duration, b: Duration) extends Schedule:
-            def next = Maybe((a, Fibonacci(b, a + b)))
-            def show = s"Schedule.fibonacci(${a.show}, ${b.show})"
+            def next(now: Instant) = Maybe((a, Fibonacci(b, a + b)))
+            def show               = s"Schedule.fibonacci(${a.show}, ${b.show})"
 
         final case class ExponentialBackoff(initial: Duration, factor: Double, maxBackoff: Duration) extends Schedule:
-            def next =
+            def next(now: Instant) =
                 val nextDelay = initial.min(maxBackoff)
                 Maybe((nextDelay, exponentialBackoff(nextDelay * factor, factor, maxBackoff)))
             def show = s"Schedule.exponentialBackoff(${initial.show}, ${formatDouble(factor)}, ${maxBackoff.show})"
         end ExponentialBackoff
 
         final case class Linear(base: Duration) extends Schedule:
-            def next = Maybe((base, linear(base + base)))
-            def show = s"Schedule.linear(${base.show})"
+            def next(now: Instant) = Maybe((base, linear(base + base)))
+            def show               = s"Schedule.linear(${base.show})"
+
+        final case class Anchored(period: Duration, offset: Duration, last: Maybe[Instant]) extends Schedule:
+            def next(now: Instant): Maybe[(Duration, Schedule)] =
+                val reference   = last.getOrElse(now)
+                val periodNanos = period.toNanos.max(1)
+
+                val elapsed  = (reference - Instant.Epoch).toNanos % periodNanos
+                val nextTime = reference - elapsed.nanos + offset
+
+                val finalTime =
+                    if nextTime <= now then
+                        val periodsToSkip = ((now - nextTime).toNanos / periodNanos) + 1
+                        nextTime + (periodsToSkip * periodNanos).nanos
+                    else nextTime
+
+                Maybe((finalTime - now, Anchored(period, offset, Present(finalTime))))
+            end next
+
+            def show =
+                val offsetStr = if offset == Duration.Zero then "" else s", ${offset.show}"
+                s"Schedule.anchored(${period.show}$offsetStr)"
+        end Anchored
 
         final case class Max(a: Schedule, b: Schedule) extends Schedule:
-            def next =
+            def next(now: Instant) =
                 for
-                    (d1, s1) <- a.next
-                    (d2, s2) <- b.next
+                    (d1, s1) <- a.next(now)
+                    (d2, s2) <- b.next(now)
```

Comment:
pre-existing: I think this should be a `fold` - otherwise you may end up discarding when one schedule produces a next.